.github/workflow-templates/go.yaml

```diff
@@ -0,0 +1,88 @@
+name: Gazebo Go
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+
+jobs:
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+          cache: true
+
+      - name: go fmt
+        run: go fmt ./...
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
+          cache: true
+
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+  vet:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+          cache: true
+
+      - name: go vet
+        run: go vet ./...
+  test:
+    needs: vet
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:5.7
+        ports:
+          - 3306
+        env:
+          MYSQL_USER: gazebo
+          MYSQL_PASSWORD: 1234
+          MYSQL_DATABASE: gazebo_db
+          MYSQL_ROOT_PASSWORD: 1234
```

Comment:
Is there a way to parameterize these, or define them as environment variables?

It might not even make sense for us to do so, but I'm making use of this PR to try to learn a thing or two.

Reply:
> Is there a way to parameterize these, or define them as environment variables?

Very good question! Yes, there are different ways to parameterize these, GitHub actions have `inputs` that you can use to configure jobs. We can an issue with environment variables when using `reusable workflows` that is stated in this PR description.  I'll need you to take a look at it so we can discuss our options.